lib/asciidoctor/opal_ext/file.rb

```diff
@@ -19,12 +19,19 @@ class File
   attr_reader :lineno
   attr_reader :path
 
-  def initialize(path, mode = 'r')
+  def initialize(path, flags = 'r')
     @path = path
     @contents = nil
     @eof = false
     @lineno = 0
-    @fd = `require('fs').openSync(path, mode)` if JAVASCRIPT_PLATFORM == 'node'
+    # binary flag is unsupported
+    binary_flag_regexp = /b/
```

Comment:
I think to read binary data with node, you set the encoding (second argument) to empty string or null. Then, you get a buffer object. You'll then have to do something with that buffer object...or trust that the next step in the chain properly converts it to base64.